Source/Controller/ThreadViewController.swift

```diff
@@ -377,25 +377,28 @@ class ThreadViewController: ContentViewController {
         let post = Post(attributedText: text, root: self.rootKey, branches: [self.branchKey])
         let images = self.galleryView.images
         let draftStore = draftStore
-        Task.detached(priority: .userInitiated) { await draftStore.save(text: text, images: images) }
-        Bots.current.publish(post, with: images) { [weak self] key, error in
-            Log.optional(error)
-            CrashReporting.shared.reportIfNeeded(error: error)
-            AppController.shared.hideProgress()
-            if let error = error {
-                self?.alert(error: error)
-            } else {
+        Task.detached(priority: .userInitiated) {
+            await draftStore.save(text: text, images: images)
+            do {
+                let messageID = try await Bots.current.publish(post, with: images)
                 Analytics.shared.trackDidReply()
-                self?.replyTextView.clear()
-                self?.replyTextView.resignFirstResponder()
-                self?.buttonsView.minimize()
-                self?.galleryView.removeAll()
-                self?.galleryView.close()
-                self?.onNextUpdateScrollToPostWithKeyValueKey = key
-                self?.load()
+                await MainActor.run {
```

Comment:
Oh, TIL about option C! Although I'm not sure if I'll use it that frequently, because A seems more expressive.

I have been using A and B in different situations. I think Option B is nice when the function executes very quickly. In this case I specifically do not want `draftStore.save(text: text, images: images)` and `await draftStore.clearDraft()` to happen on the Main Actor because they are slow and in my testing it didn't seem like NSKeyedUnarchiver and NSKeyedArchiver correctly yield while they are writing to disk. So I didn't want to make the whole function run on the main actor. Does that seem like the best option to you?